policy/modules/system/getty.te

```diff
@@ -84,6 +84,7 @@ term_setattr_unallocated_ttys(getty_t)
 term_setattr_console(getty_t)
 
 auth_rw_login_records(getty_t)
+auth_use_nsswitch(getty_t)
```

Comment:
This seems strange, do you know why this is happening in the first place?

Reply:
Sorry I should have added more context.

This is actually due to interactions with io.systemd.DynamicUsers and userdb.

Adding auth_use_nsswitch(getty_t) fixes the following avc errors on a Buildroot-based system :

audit: type=1400 audit(1610032673.915:3): avc:  denied  { read } for  pid=75 comm="agetty" name="userdb" dev="tmpfs" ino=812 scontext=system_u:system_r:getty_t tcontext=system_u:object_r:systemd_userdb_runtime_t tclass=dir permissive=1

audit: type=1400 audit(1610032673.915:3): avc:  denied  { open } for  pid=75 comm="agetty" path="/run/systemd/userdb" dev="tmpfs" ino=812 scontext=system_u:system_r:getty_t tcontext=system_u:object_r:systemd_userdb_runtime_t tclass=dir permissive=1

audit: type=1400 audit(1610032673.959:4): avc:  denied  { getattr } for  pid=75 comm="agetty" path="/run/systemd/userdb" dev="tmpfs" ino=812 scontext=system_u:system_r:getty_t tcontext=system_u:object_r:systemd_userdb_runtime_t tclass=dir permissive=1

audit: type=1400 audit(1610032673.979:5): avc:  denied  { search } for  pid=75 comm="agetty" name="userdb" dev="tmpfs" ino=812 scontext=system_u:system_r:getty_t tcontext=system_u:object_r:systemd_userdb_runtime_t tclass=dir permissive=1

audit: type=1400 audit(1610032673.979:5): avc:  denied  { write } for  pid=75 comm="agetty" name="io.systemd.DynamicUser" dev="tmpfs" ino=814 scontext=system_u:system_r:getty_t tcontext=system_u:object_r:systemd_userdb_runtime_t tclass=sock_file permissive=1

Reply:
It seems like this shouldn't be necessary since AFAICS agetty only does host/domainname lookups. Does agetty break over those denials? If not, maybe we should split up `auth_use_nsswitch()` into different nsswitch services.